File: lib/go/src/github.com/eliudarudo/go-cli-app/player/input_mod.go
package player

import (
	"fmt"
	"strconv"
	"strings"
	"time"
)

// GetSelection function
func GetSelection() string {
	fmt.Println("==================================================")
	fmt.Print("Enter a selection (Q to quit): ")

	var selection string
	fmt.Scanln(&selection)

	return strings.ToUpper(selection)
}

// GetSong function
func GetSong() *Song {
	var title, artist, ratingStr string
	var rating int

	for {
		fmt.Print("Enter song name: ")
		fmt.Scanln(&title)
		if StringIsValid(title) {
			break
		}
	}

	time.Sleep(50 * time.Millisecond)

	for {
		fmt.Print("Enter song artist: ")
		fmt.Scanln(&artist)
		if StringIsValid(artist) {
			break
		}
	}

	for {
		fmt.Print("Enter your rating: ")
		fmt.Scanln(&ratingStr)
		ratingVal, err := strconv.Atoi(ratingStr)
		if err != nil {
			fmt.Println("Please enter a number")
		} else {
			rating = ratingVal
			break
		}
	}

	song := &Song{
		Title:  title,
		Artist: artist,
		Rating: rating,
	}

	// Returns reference to the song object created
	return song
}

File: server/src/lib/java/src/com/eliudarudo/cliapp/player/InputMod.java
package com.eliudarudo.cliapp.player;

import java.util.Scanner;

public class InputMod {
    public InputMod() {}

    public static String getSelectionWithPrompt(String prompt) {
        Scanner scanner = new Scanner(System.in);
        System.out.print(prompt);

        try {
            String selection = scanner.nextLine().toUpperCase();
            return selection;
        } catch(Exception e) {
            return "";
        }
    }

    public static String getSelection() {
        System.out.println("==================================================");
        String selection = getSelectionWithPrompt("Enter a selection (Q to quit): ");
        return selection;
    }

    public static Song getSong() {
        String title = "";
        String artist = "";
        String ratingString = "";
        int rating = 0;

        while(true) {
            title = getSelectionWithPrompt("Enter song name: ").trim();
            if(Utility.stringIsValid(title)) {
                break;
            }
        }

        while(true) {
            artist = getSelectionWithPrompt("Enter song artist: ").trim();
            if(Utility.stringIsValid(artist)) {
                break;
            }
        }

        while(true) {
            try {
                ratingString = getSelectionWithPrompt("Enter your rating: ").trim();

                if(Utility.stringIsValid(ratingString)) {
                    rating = Integer.parseInt(ratingString);
                    break;
                } else {
                    throw new Exception("");
                }

            } catch(Exception e) {
                System.out.println("Please enter a number");
            }
        }

        Song song = new Song(title, artist, rating);
        return song;
    }
}
